src/App.tsx

```diff
@@ -1,39 +1,52 @@
-import React from 'react';
+import { Component } from 'react';
+import { Clock } from './Clock';
 import './App.scss';
 
-function getRandomName(): string {
-  const value = Date.now().toString().slice(-4);
+const getRandomName = (): string => `Clock-${Date.now().toString().slice(-4)}`;
 
-  return `Clock-${value}`;
-}
+type ClockState = {
+  hasClock: boolean,
+  clockName: string,
+};
+
+export class App extends Component<{}, ClockState> {
+  state: Readonly<ClockState> = {
+    clockName: 'Clock-0',
+    hasClock: true,
+  };
+
+  componentDidMount() {
+    document.addEventListener('contextmenu', this.handleDocContextMenu);
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+    document.addEventListener('contextmenu', (event) => {
+      event.preventDefault();
+    });
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+    document.addEventListener('click', this.handleDocClick);
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+    window.setInterval(() => {
```

Comment:
clear this interval

Reply:
I'm not following, why? That's the interval for the optional task, it logs clock renames